### PR TITLE
emacs: adapt substituteInPlace invocation to sync with upstream

### LINKED ISSUE #251: `emacsGit` and the lot haven't updated since 9/21
Should by fixed by https://github.com/NixOS/nixpkgs/pull/193621. Posting to obviate anyone else's need to look into this.

### PR DESCRIPTION
**Pull request by leungbk**
###### Description of changes

Fixes https://github.com/nix-community/emacs-overlay/issues/251.

The emacsGit attribute and other natively-compiled master-based attributes in the Emacs overlay [have been failing to update](https://github.com/nix-community/emacs-overlay/actions/runs/3162774493/jobs/5149686344#step:6:25439) for more than a week because [commit 97b928ce09d6034ebcb541fb548e5d4862302add](https://git.savannah.gnu.org/cgit/emacs.git/commit/?id=97b928ce09d6034ebcb541fb548e5d4862302add) in Emacs messed up the old `substituteInPlace`.

The `substituteInPlace` invocation added in this patch occurs after the definition of `native-comp-driver-options` in both Emacs 28 and 29, so this is safe. I tested by building a native-compiled Emacs (each of 28.2 and a version of 29 newer than what is presently in the overlay) with the `vterm` package installed. I noticed that I needed to install external packages alongside a native-compiled Emacs containing the upstream commit to force a Nix build without this PR to fail.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by leungbk** [comment]
openblas 3.21 isn't building right now on aarch64-linux, causing the test failure there.

**Review by Atemu — CHANGES_REQUESTED** [review]
Since this has silently failed once already, could we make this a patch instead? That will blow up the derivation right from the start when the patch doesn't apply anymore.

File: pkgs/applications/editors/emacs/generic.nix [hunk 1]
```diff
@@ -67,7 +67,31 @@ let emacs = (if withMacport then llvmPackages_6.stdenv else stdenv).mkDerivation
...
 
-  patches = patches fetchpatch;
+  patches = patches fetchpatch ++
+            # TODO: This patch is written awkwardly because at the
+            # time of writing, the defcustom for
+            # native-comp-driver-options is set to a different value
+            # in Emacs 28.2 as compared to Emacs 28-unstable and Emacs
+            # 29.  Upon removing Emacs 28 from Nixpkgs, it is
+            # recommended that we patch the defcustom directly,
+            # instead of finding a suitable place afterward to setq.
```
Comment (Atemu) [inline pkgs/applications/editors/emacs/generic.nix:77]: The point of the patch is to break when upstream changes something. This way there is no point in using a patch.

If we're going to do patches, we need different patches for different versions.

**Review by Atemu — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
> Decided to target `master` since the mass rebuilds are simply Emacs packages.

Still we are rebuilding **a lot** of packages.

**Comment by Atemu** [comment]
@SuperSandro2000 those are emacsPackages. Mostly tiny packages with a handful of elisp files that need to be byte-compiled, native-compiled and installed.

**Pull request closed without merging**